libyul/AsmAnalysis.cpp

```diff
@@ -772,10 +772,27 @@ bool AsmAnalyzer::validateInstructions(evmasm::Instruction _instr, SourceLocatio
 			"PC instruction is a low-level EVM feature. "
 			"Because of that PC is disallowed in strict assembly."
 		);
+	else if (!m_eofVersion.has_value() && (
+		_instr == evmasm::Instruction::EXTCALL ||
+		_instr == evmasm::Instruction::EXTDELEGATECALL ||
+		_instr == evmasm::Instruction::EXTSTATICCALL
+	))
+	{
+		m_errorReporter.typeError(
+			4328_error,
+			_location,
+			fmt::format(
+				"The \"{instruction}\" instruction is {kind} VMs.",
+				fmt::arg("instruction", boost::to_lower_copy(instructionInfo(_instr, m_evmVersion).name)),
+				fmt::arg("kind", "not available in legacy bytecode")
```

Comment:
Since we're keeping this bit after all, I'll add one more suggestion:

```suggestion
				"The \"{}\" instruction is only available on EOF.",
				fmt::arg("instruction", boost::to_lower_copy(instructionInfo(_instr, m_evmVersion).name))
```

This is more concise and also the `kind` makes this confusing to me every time I look at this message because it's not kind of something, it's just a part of the message.

Also I'd refer to EOF explicitly. "Legacy" can be many things and we're generally overusing it in the compiler. Even in this context there is one other thing it could refer to - the older EVMs we deprecated support for. Saying "EOF" removes the ambiguity. Since this code is here solely to give a more specific error message, clarity is the whole point :)

Would be nice to change the one below to match too.